kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java

```diff
@@ -30,16 +30,38 @@ protected BaseContributor(BaseContributorBuilder<T, S> builder) {
         shortNameToInstanceBuilder = builder.build();
     }
 
+    @Override
+    public boolean contributes(String shortName) {
+        return shortNameToInstanceBuilder.containsKey(shortName);
+    }
+
+    @Override
+    public ConfigurationDefinition getConfigDefinition(String shortName) {
+        return shortNameToInstanceBuilder.get(shortName).configurationDefinition();
+    }
+
+    @SuppressWarnings("removal")
     @Override
     public Class<? extends BaseConfig> getConfigType(String shortName) {
-        InstanceBuilder<T, S> instanceBuilder = shortNameToInstanceBuilder.get(shortName);
-        return instanceBuilder == null ? null : instanceBuilder.configClass;
+        final ContributionDetails<?, ?> contributionDetails = shortNameToInstanceBuilder.get(shortName);
+        if (contributionDetails != null) {
+            return contributionDetails.configurationDefinition().configurationType();
+        }
+        else {
+            return null;
```

Comment:
I think we should switch to throw `IllegalArgumentException` now that there is the `boolean contributes(String shortName)`. Keep the null in it's cage :grinning: WDYT

Reply:
I'm all for caging `null`

Reply:
ah, backwards compatibility :thinking: maybe just the new method should throw and we could put a note on `getInstance` that it will start throwing in future?

Reply:
Can we play the 0.2 -> 0.3 evolving API card?